### ISSUE TITLE
Anchor wrapped in partial does not work as expected

### ISSUE THREAD

**Issue by ramik91** [ractivejs/ractive#3271]
## Description:
I develop an app that uses anchor as a mount point for my components. Everything works just fine untill I use this approach inside a partial of a parent component. Following behavior appears if I do so.

1) children that are attached to my anchor gets rendered correctly for the first time but changing data via set method does not propagate down the component tree anymore.

2) on-click handlers gets called twice.

## Versions affected:
0.10.9, 0.10.10, 1.0.0

## Platforms affected:
Tested on MS Windows 10 (Chrome, FF, Edge)

## Reproduction:
Here is a [code snippet](https://ractive.js.org/playground/#N4IgLiBcIDwCYEsBuACBcC8ByATgezzCwD55liQAaEAZyhAGM8A7GsFAYQBs8GBrFBhQAlAIYMwyAKYA6KQA8wU5nAAUwADrMUKJQFsADl1FLIKLDBoHRzYsGC6Ee2WDwAZXqK5SAKk6kAymA4CMwA5qoAlCgAvjEwAPRWNsRYWjGRANxaWkys7ABqCFIA7oIi4pJIsgpKKupaOkyGLMpgNGaa2jooDDz8Zty8fI2xlKNwJqJmUYLEKA3dOhpgks5mzKUoACImUlE5YBnj3WFSYH7Os109ugAWCDQyNOeqWGtSWJQom2W7SlEsqNlkteiwaHhvDIeBEsEUth80DRzN8wA8nmcwG8PlhIkDujETjp9EY9mYAAaHFZkVB9UQ0GgYDQgJDFErM4hUsDUgBGAFdViwUCwALR9BD8JkgAACMkxl32kQ5AHFzo5nIl+YLbFzqX1huqpBh7IbYjEUAlOcwVtSEogkFaVpTmBlstbmHk2CgAIIeu54HDlMQSaRyRTKNQ3YXMTrAtDMBBY6JR26e9jNAytZjsIS-FDwkpRN23EEl+6PGQmMDiO4cB5cNQZrNgb4Oas4TFmLB6PB87MABTwoSIsXxtxio0JoxJxlM5hgAGIe32wIPh5a0i78blwew3KIAJ69nMVEPVMN1SOjJubbMdFAplA2Bj+nBmX0vgOTokoaw4SRePej48ngcAHl2MDPq+G7ftOUiGLOUgUlyMB3AAjL0xgMlKkhgN4HLegYBgoCUiZ3E+foBok6GOtyYA0ph9KMsyIFgRyJrEKxB5mok9q0c6ro5B6u5PkR5R5vuR4CosOhSFwXYLvghCboJzBUIw9DqXcUCgDg9CIWwIBxEAA).

**Comment by ghost** [comment]
[I've simplified the issue](https://ractive.js.org/playground/#N4IgLiBcIDwCYEsBuACBcC8ByATgezzCwD55liQAaEAZyhAGM8A7GsFAYQBs8GBrFBhQAlAIYMwyAKYA6KQA8wU5nAAUwADrMUKJQFsADl1FLIKAAZaNYa2STEAKgj1S027rz5ngwXc9cAvgEwAPSI9lZglswBAJQA3FbMTKzsAGoIUgDugiLikkiyCkoq6lo6TIYsymA03uU6nDz8DQGUDXAmomaqsYLEKGXajZIuZgCyJgAWMjiiKnh6vZFx7cP6RiZSZtGNKHaO-m4oSJlZ3r6jgcFh5A06MB78fi4YGiA+L9fvKCHE9-sAEYAVzAYBYKBYAFoGFwEPw3iAAAJgKYIGgyADmUjAThcyxAxAA4jivqEQWCWP9mNZzGsdNjcf5eihNMMdKj0TIaDjVFgrlhKChJqjZvM4ItegkAQCUjQ8FxZDxMXyMtkvmgaChBbo0RjGXyBbFpcMAlo4okackWGwUAAZUQATzwoNyYgk0jkimUajZHKkhmMpgskVsUwAjChYaIaDREZIwIr3sQAIIGAwoLIIVEoeYMKZ4HChCPU6y2cJR4yxxGAvBwR3JmAAYj0LuYYAACngEO2-qFwqWomtrCwe9mWX7GnL2JUDNV27lmOq1VlevEUADORiTGBxFMOGiuGpZ-OwELLqIcIyzFhW8D212e0QUBaN6a1haktPc+nF+qHc6oJDDoUhcDeTb4IQWDmiaVCMPQcFTFAoA4PQQZSGwIBBEAA). It has nothing to do with partials but more of how components work when dynamically mounted. In this case, when dynamically mounting `View`, it doesn't sync `time` to `Clock`. The log still happens twice though, so there is something weird in there.

In the following example, if you replace `<#mountPoint/>` with `<View/>` (register `View` as a component of `Layout`), everything starts working.

```js
const Clock = Ractive.extend({
  template: `
    <div>Time in Clock: {{ time }}</div>
  `
});

const View = Ractive.extend({
  components: {
    Clock
  },
  data: () => ({
    time: Math.random()
  }),
  template: `
    <div>Time in view: {{ time }}</div>
    <Clock time="{{ time }}" />
    <button on-click="@this.getTime()">Get time</button>
  `,
  getTime() {
    this.set('time', Math.random());

    console.log('View time is ', this.get('time'));
  }
});

const Layout = Ractive.extend({
  template: `
    <h1 class="title">App with anchor</h1>
    <div class="body"><#mountPoint/></div>
  `,
  oninit() {
    const component = new View();
    this.attachChild(component, {
      target: 'mountPoint'
    });
  },
});

const app = new Layout({
  el: '#root'
});
```

**Comment by evs-chris** [comment]
This is actually two separate semi-related bugs that I've identified, but it may take me a bit to squish the broken bindings side. What happens is the instance unbinds and binds again when it's attached to the anchor, which causes the `Clock` instance to unbind and tear down, which leaves it in a dead state even though it is rendered.

**Comment by evs-chris** [comment]
I forgot that anchors are inherently async, so getting a test case that would break took a bit. I think I've got it now, and the fix should be on edge, v0.10.11-dev and v1.0.1-dev as soon as travis finishes travising.
